distribution/registry.go

```diff
@@ -18,6 +19,31 @@ import (
 	"golang.org/x/net/context"
 )
 
+// ImageTypes represents the schema2 config types for images
+var ImageTypes = []string{
+	schema2.MediaTypeConfig,
```

Comment:
I think we also have to allow `application/octet-stream` to be compatible with a bug in previous versions. It's probably also a good idea to accept an empty string as an image type, though I don't know of any configuration that will produce this.

The code appears to be changing to whitelist particular mediatypes instead of blacklisting incompatible ones. This is probably the right model, but I wonder if we'll discover some unexpected media types in rare manifests produced by third parties. Something to be aware of.